.github/workflows/cppcheck-differential.yaml

```diff
@@ -0,0 +1,83 @@
+name: cppcheck-differential
+
+on:
+  pull_request:
+
+jobs:
+  cppcheck-differential:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y wget
+
+      # cppcheck from apt does not yet support --check-level args, and thus install from snap
+      - name: Install Cppcheck from snap
+        run: |
+          sudo snap install cppcheck
+
+      # Download the cppcheck suppression file
+      - name: Download cppcheck suppression file
+        run: |
+          wget https://raw.githubusercontent.com/autowarefoundation/autoware.universe/main/.cppcheck_suppressions -O .cppcheck_suppressions
+
+      - name: Remove exec_depend
+        uses: autowarefoundation/autoware-github-actions/remove-exec-depend@v1
+
+      - name: Get modified packages
+        id: get-modified-packages
+        uses: autowarefoundation/autoware-github-actions/get-modified-packages@v1
+
+      # Display the modified packages
+      - name: Show modified packages
+        run: |
+          echo "Modified packages:"
+          echo "${{ steps.get-modified-packages.outputs.modified_packages }}"
+
+      - name: Get modified files
+        id: get-modified-files
+        uses: tj-actions/changed-files@v44
+        with:
+          files: |
+            **/*.cpp
+            **/*.hpp
+
+      # Display the modified files
+      - name: Show modified files
+        run: |
+          echo "Modified files:"
+          echo "${{ steps.get-modified-files.outputs.all_changed_files }}"
+
+      - name: Run Cppcheck on changed files
+        continue-on-error: true
+        id: cppcheck
+        run: |
+          files=$(echo ${{ steps.get-modified-files.outputs.all_changed_files }})
+          if [ -n "$files" ]; then
+            echo "Running Cppcheck on changed files: $files"
+            cppcheck --inline-suppr --enable=all --inconclusive --check-level=exhaustive --error-exitcode=1 --suppressions-list=.cppcheck_suppressions $files 2> cppcheck-report.txt
+          else
+            echo "No C++ files changed."
+            touch cppcheck-report.txt
+          fi
+        shell: bash
+
+      - name: Show cppcheck-report result
+        run: |
+          cat cppcheck-report.txt
+
+      - name: Upload Cppcheck report
+        uses: actions/upload-artifact@v4
+        with:
+          name: cppcheck-report
+          path: cppcheck-report.txt
+
+      - name: Fail the job if Cppcheck failed
+        if: steps.cppcheck.outcome == 'failure'
+        run: exit 1
```

Comment:
Is it possible that if there is no difference in the cpp/hpp file, an error will occur because there is no file to cppcheck?

Reply:
"Run Cppcheck on changed files" is set to "continue-on-error: true" because the error is ignored, "if: steps.cppcheck.outcome == 'failure'" to catch the error occurrence.
If there is no difference in the file, it is considered a success and therefore not an error.

Reply:
Appreciate the response. This is the first time I have learned of such a function. I have checked the documentation and it does not seem to cause the problem I was concerned about.
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error